packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -164,50 +179,59 @@ export class SOQLEditorInstance {
         this.updateTextDocument(this.document, soql);
         break;
       }
+      case MessageType.UI_TELEMETRY: {
+        const { unsupported } = e.payload as TelemetryModelJson;
+        const hasUnsupported = Array.isArray(unsupported)
+          ? unsupported.length
+          : unsupported;
+        if (hasUnsupported) {
+          trackErrorWithTelemetry(
+            'syntax_unsupported',
+            JSON.stringify(e.payload)
+          ).catch(console.error);
+          const message = nls.localize('info_syntax_unsupported');
+          channelService.appendLine(message);
+        }
+        break;
+      }
       case MessageType.SOBJECT_METADATA_REQUEST: {
-        this.retrieveSObject(e.payload as string).catch(() => {
-          channelService.appendLine(
-            `An error occurred while handling a request for object metadata for the ${e.payload} object.`
-          );
-        });
+        this.retrieveSObject(e.payload as string);
         break;
       }
       case MessageType.SOBJECTS_REQUEST: {
-        this.retrieveSObjects().catch(() => {
-          channelService.appendLine(
-            `An error occurred while handling a request for object names.`
-          );
-        });
+        this.retrieveSObjects();
         break;
       }
       case MessageType.RUN_SOQL_QUERY: {
-        this.handleRunQuery().catch(() => {
-          channelService.appendLine(
-            `An error occurred while running the SOQL query.`
-          );
-        });
+        this.handleRunQuery();
         break;
       }
       default: {
-        console.log('message type is not supported');
+        const message = nls.localize('error_unknown_error', e.type);
+        channelService.appendLine(message);
+        trackErrorWithTelemetry('message_unknown', e.type).catch(console.error);
```

Comment:
An example of this message.  This would basically only happen if the soql-builder-ui and soql-extension versions weren't compatible.

<img width="1199" alt="unknown-message" src="https://user-images.githubusercontent.com/599418/101825613-0fb24900-3aeb-11eb-8a64-1e5c629d63d8.png">